src/register-stories.js

```diff
@@ -0,0 +1,63 @@
+const Vue = require('vue')
+
+const capitalize = ([first, ...rest], lowerRest = false) =>
+  first.toUpperCase() + (lowerRest ? rest.join('').toLowerCase() : rest.join(''));
+
+const camelCase = str => {
+  let s =
+    str &&
+    str
+      .match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g)
+      .map(x => x.slice(0, 1).toUpperCase() + x.slice(1).toLowerCase())
+      .join('');
+  return s.slice(0, 1).toLowerCase() + s.slice(1);
+};
+
+function registerStories (req, fileName, sb, plugins) {
+  const {
+    action,
+    withNotes,
+    text,
+    boolean,
+    number,
+    color,
+    object,
+    array,
+    select,
+    date
+  } = plugins
+  const componentConfig = req(fileName)
+  const componentName = capitalize(
+    camelCase(
+      fileName
+        .replace(/^\.\/_/, '')
```

Comment:
I would replace the `_` with a `[\W_]*?` to be more generic to different file naming schemes. The latter matches any number of non-word characters or underscores that might prefix a filename.

Reply:
Good call!